scala2-functional-program-design/week5/calculator/src/main/scala/calculator/Calculator.scala

```diff
@@ -11,11 +11,20 @@ final case class Divide(a: Expr, b: Expr) extends Expr
 object Calculator extends CalculatorInterface {
   def computeValues(
       namedExpressions: Map[String, Signal[Expr]]): Map[String, Signal[Double]] = {
-    ???
+    namedExpressions.map{
+      case (name, sig) => (name, Signal[Double](eval(sig.apply(), namedExpressions)))
```

Comment:
```suggestion
      case (name, sig) => (name, Signal[Double](eval(sig(), namedExpressions)))
```

Reply:
It wouldn't work since "eval" method requires "expression" and not "signal" type